File: app/src/main/java/com/carterchen247/camerademo/ResultActivity.java
package com.carterchen247.camerademo;

import android.graphics.Bitmap;
import android.net.Uri;
import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import android.view.ViewTreeObserver;
import android.view.WindowManager;
import android.widget.ImageView;

import com.carterchen247.camerakit.util.ImageUtil;

/**
 * Created by kazaf on 2018/1/4.
 */

public class ResultActivity extends AppCompatActivity {

    ImageView img;
    private static Bitmap mBitmap;
    String path;
    boolean isFrontCamera;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_blank);
        img = (ImageView) findViewById(R.id.img);
        path = getIntent().getExtras().getString(Constant.FILE);
        isFrontCamera = getIntent().getBooleanExtra(Constant.IS_FRONT_CAMERA, false);

        img.getViewTreeObserver().addOnPreDrawListener(new ViewTreeObserver.OnPreDrawListener() {
                            @Override
                            public boolean onPreDraw() {
                                setImageBitmap();
                                img.getViewTreeObserver().removeOnPreDrawListener(this);
                                return true;
                            }
                        });

        getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON | WindowManager.LayoutParams.FLAG_FULLSCREEN);
    }

    /** Sets bitmap to ImageView widget */
    private void setImageBitmap() {
        final int width = img.getMeasuredWidth();
        final int height = img.getMeasuredHeight();

        // TODO IMPROVE MEMORY USAGE HERE, ESPECIALLY ON LOW-END DEVICES.
        if (mBitmap == null)
            mBitmap = ImageUtil.getRotatedBitmap(Uri.parse(path).getPath(), width, height, isFrontCamera);

        img.setImageBitmap(mBitmap);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (mBitmap != null && !mBitmap.isRecycled()) {
            try {
                mBitmap.recycle();
                mBitmap = null;
            } catch (Throwable t) {
                t.printStackTrace();
            }
        }
    }
}

File: app/src/main/java/com/carterchen247/camerademo/ResultActivity.kt
package com.carterchen247.camerademo

import android.net.Uri
import android.os.Bundle
import android.view.WindowManager
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.doOnPreDraw
import com.carterchen247.camerakit.util.ImageUtil
import kotlinx.android.synthetic.main.activity_result.*

class ResultActivity : AppCompatActivity() {

    companion object {
        const val FILE_PATH = "FILE_PATH"
        const val IS_FRONT_CAMERA = "IS_FRONT_CAMERA"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_result)
        window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON or WindowManager.LayoutParams.FLAG_FULLSCREEN)

        val path = intent.getStringExtra(FILE_PATH)
        val isFrontCamera = intent.getBooleanExtra(IS_FRONT_CAMERA, false)

        capturedImg.doOnPreDraw {
            loadCapturedImage(path, isFrontCamera)
        }
    }

    private fun loadCapturedImage(path: String?, isFrontCamera: Boolean) {
        capturedImg.run {
            val bitmap = ImageUtil.getRotatedBitmap(Uri.parse(path).path, width, height, isFrontCamera)
            setImageBitmap(bitmap)
        }
    }
}